### PR TITLE
Added caching feature

### PR DESCRIPTION
**Pull request by aico60**

### CODE REVIEW

File: ProjectB/ProjectB/Infrastructure/CacheFilter.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿using Microsoft.Extensions.Caching.Memory;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ProjectB.Infrastructure
+{
+    public class CacheFilter
```
Comment (it-kyrylo) [inline ProjectB/ProjectB/Infrastructure/CacheFilter.cs]: Add ICacheFilter for this class

File: ProjectB/ProjectB/Services/HotelService.cs [hunk 2]
```diff
@@ -3,56 +3,87 @@
...
 namespace ProjectB.Services
 {
     public class HotelService : IHotelService
     {
         private IHotelClients hotelClients;
         private IMapper mapper;
+        private CacheFilter cacheFilter;
 
-        public HotelService(IHotelClients hotelClients, IMapper mapper)
+        public HotelService(IHotelClients hotelClients, IMapper mapper, CacheFilter cacheFilter)
```
Comment (it-kyrylo) [inline ProjectB/ProjectB/Services/HotelService.cs]: same use ICacheFilter instead of CacheFilter

**Review by it-kyrylo — CHANGES_REQUESTED** [review]
I think the only good place for caching it is `GetHotelDetailsById` so please remove all caching stuff from `GetDestinationIdAsync` and `GetHotelsByDestinationIdAsync`

also please take a look at how I implemented ICacheFilter, and try to run it locally and fix all problems if you find them, and I swear you will find them ;)

**Comment by aico60** [comment]
Changes done and waiting for review.

**Comment by vladimirdonev** [comment]
Approve!

**Review by it-kyrylo — APPROVED** [review]

**Comment by VeselinMetodiev** [comment]
I Approve! Good job, Aykut!

**Pull request merged**